File: Segundo Periodo/Arquitetura de computadores/2023-2_arq1_guia_01/exercicio01/Guia01.java
import java.util.Scanner;
public class Guia01 {
    
    public boolean isNumber(String test){
        int quantity = 0;
        for(int i = 0; i < test.length(); i++){
            if('0' <= test.charAt(i) && test.charAt(i) <= '9'){
                quantity++;
            }
        }
        if(quantity == test.length()){
            return true;
        }else{
            return false;
        }
    }

    public int toInt(String decimal){
        int dec = 0;
        boolean test = isNumber(decimal);
        if(test){
            for(int i = 0; i < decimal.length(); i++){
                dec = (int) Math.pow(10,decimal.length() - 1 - i) * (decimal.charAt(i) - 48);
            }
            return dec;
        }else{
            System.out.println("Ha uma letra que nao eh um numeral!");
            return 0;
        }
    }

    public String toBinario(int num){
        String convertido = "";
        while(num > 0){
            convertido = (num % 2) + convertido;
            num /= 2;
        }
        return convertido;
    }

    public boolean isEquals(String one, String two){
        if(one.length() == two.length()){
            for(int i = 0; i < one.length(); i++){
                if(one.charAt(i) != two.charAt(i)){
                    return false;
                }
            }
        }else{
            return false;
        }
        return true;
    }

    public void conversor() {
        String num = "";
        int number = 0;
        Scanner scanner = new Scanner(System.in);
        while(num != "end"){
            System.out.println("Digite end para sair do loop!");
            System.out.print("Digite o numero que deseja converter para binario: ");
            num = scanner.nextLine();
            if(isEquals(num, "end")){
                break;
            }
            number = toInt(num);
            System.out.println(toBinario(number));
        }
    }
    
}

File: Segundo Periodo/Arquitetura de computadores/2023-2_arq1_guia_01/exercicio01/Guia01.py
class ToBinary:
    def __init__(self):
        self.strNum = ''
        self.intNum = ''

    def isNumber(self):
        numb = 0
        for i in range(len(self.strNum)):
            if '0' <= self.strNum and self.strNum <= '9':
                numb += 1
        if numb == len(self.strNum):
            return True
        else:
            return False

    def toBinary(self):
        if self.isNumber():
            numtoInt = int(self.strNum)
            if numtoInt == 0:
                self.intNum = '0'
            while numtoInt > 0:
                self.intNum = f'{numtoInt % 2}{self.intNum}'
                numtoInt = numtoInt // 2

        else:
            print('Entrada invalida!')
    
    def setNumber(self,num):
        self.strNum = num
        self.intNum = ''
        self.toBinary()

    def getNumber(self):
        return self.intNum

converter = ToBinary()

while(True):
    number = input('Digite end para sair do loop\nDigite o numero que quer converter para binario: ')
    if number == 'end':
        break
    converter.setNumber(number)
    print(converter.getNumber())
